api_factory/schema/api.py

```diff
@@ -219,37 +281,34 @@ def _get_methods(self, methods: List[descriptor_pb2.MethodDescriptorProto],
                 sequence of protobuf method objects.
             address (~.metadata.Address): An address object denoting the
                 location of these methods.
-            info (dict): The appropriate slice of proto comments
-                corresponding to these methods.
+            path (Tuple[int]): The source location path thus far, as understood
+                by ``SourceCodeInfo.Location``.
 
         Returns:
             Mapping[str, ~.wrappers.Method]: A ordered mapping of
                 :class:`~.wrappers.Method` objects.
         """
         # Iterate over the methods and collect them into a dictionary.
         answer = collections.OrderedDict()
-        for method_pb, i in zip(methods, range(0, sys.maxsize)):
-            types = method_pb.options.Extensions[lro_pb2.types]
-            answer[method_pb.name] = wrappers.Method(
-                input=self.messages[method_pb.input_type.lstrip('.')],
-                lro_metadata=self.messages.get(types.lro_metadata_type, None),
-                lro_payload=self.messages.get(types.lro_return_type, None),
-                method_pb=method_pb,
+        for meth_pb, i in zip(methods, range(0, sys.maxsize)):
+            types = meth_pb.options.Extensions[operations_pb2.operation_types]
+            answer[meth_pb.name] = wrappers.Method(
+                input=self.all_messages[meth_pb.input_type.lstrip('.')],
+                lro_metadata=self.all_messages.get(types.metadata, None),
+                lro_payload=self.all_messages.get(types.response, None),
```

Comment:
This will cause a silent failure if the payload message is not part of the import tree. Since it's a breaking for the generated client to fix the lro method, would we want to consider adding something to ensure that if a non empty payload type string is provided, a message type must be found?

Reply:
I agree with this, but it is not a change from the status quo. Can I fix that in a follow-up? (I have another PR I need to base off this one.)

Reply:
Lets fix this in a followup. Also, lets make sure that the issue is tracked.